### ISSUE TITLE
Incorrect index accessor against optional property

### ISSUE THREAD

**Issue by danielrbradley** [pulumi/pulumi#11495]
### What happened?

Attempting to access a nested property within an optional array property generates invalid TypeScript code.

### Steps to reproduce

Run `pulumi convert --language typescript --out typescript` for the following YAML program.

```yaml
variables:
  appName: my-app
resources:
  repository:
    type: awsx:ecr:Repository
  image:
    type: awsx:ecr:Image
    properties:
      repositoryUrl: ${repository.url}
      path: "./app"
  cluster:
    type: eks:Cluster
  clusterProvider:
    type: pulumi:providers:kubernetes
    properties:
      kubeconfig: ${cluster.kubeconfig}
      enableServerSideApply: true
  deployment:
    type: kubernetes:apps/v1:Deployment
    properties:
      metadata:
        labels:
          appClass: ${appName}
      spec:
        replicas: 2
        selector:
          matchLabels:
            appClass: ${appName}
        template:
          metadata:
            labels:
              appClass: ${appName}
          spec:
            containers:
              - name: ${appName}
                image: ${image.imageUri}
                ports:
                  - name: http
                    containerPort: 80
    options:
      provider: ${clusterProvider}
  service:
    type: kubernetes:core/v1:Service
    properties:
      metadata:
        labels:
          appClass: ${appName}
      spec:
        type: LoadBalancer
        selector:
          appClass: ${appName}
        ports:
          - port: 80
            targetPort: http
    options:
      provider: ${clusterProvider}
outputs:
  url: ${service.status.loadBalancer.ingress[0].hostname}
```

### Expected Behavior

Output should be generated as:

```typescript
export const url = service.status.apply(status => status?.loadBalancer?.ingress?.[0]?.hostname);
```

### Actual Behavior

Output should be generated as:

```typescript
export const url = service.status.apply(status => status?.loadBalancer?.ingress?[0]?.hostname);
```

Required change is a single dot:

```diff
-export const url = service.status.apply(status => status?.loadBalancer?.ingress?[0]?.hostname);
+export const url = service.status.apply(status => status?.loadBalancer?.ingress?.[0]?.hostname);
```

### Output of `pulumi about`

```
CLI          
Version      3.48.0
Go Version   go1.19.3
Go Compiler  gc

Plugins
NAME        VERSION
awsx        unknown
eks         unknown
kubernetes  unknown
yaml        unknown

Host     
OS       darwin
Version  13.0.1
Arch     arm64

Current Stack: dev

Found no resources associated with dev

Found no pending operations associated with dev

Backend        
Name           pulumi.com
URL            https://app.pulumi.com/daniel-pulumi
User           daniel-pulumi
Organizations  daniel-pulumi, pulumi
```

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by justinvp** [comment]
I believe this is a dupe of #8068 which was fixed 2 days ago by #11477

**Comment by danielrbradley** [comment]
Bah! I searched, but guess I missed it because it was closed 😆  Thanks!
